akd/src/storage/transaction.rs

```diff
@@ -87,99 +82,88 @@ impl Transaction {
 
     /// Start a transaction in the storage layer
     pub async fn begin_transaction(&self) -> bool {
```

Comment:
can we now refactor out the async signature too since we're using atomics? it'll remove the `Poll` in tokio since there's no async work iirc